keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -1506,32 +1471,155 @@ KEPs.  The first change doesn’t present any user visible change, and if
-nodes to accommodate this pod. Meanwhile, the scheduler will evaluate the
-pod-level resource requests but may be unable to find a suitable node to fit the
-pod. Consequently, the pod will not be scheduled. While this behavior is
-acceptable for the alpha implementation, it is anticipated that Cluster
-Autoscaler support will be addressed in the Beta phase with pod resource
-requirements surfaced in a helper library/function that autoscalers can use to
-make autoscaling decisions.
+* [Scoped for GA] Phase 2 of improving recommendation algorithm
+  Pod-Level Resources allows pod-level limits to be greater than aggregated
+  container limits to allow the containers to share idle resources among each other.
+  Integrating this functionality with VPA necessitates the development of a 
+  complex new recommendation algorithm. Concepts such as proportionate pod
+  and container level recommendations have been proposed and 
+  require further discussion.
 
-#### [Scoped for Beta] Support for Windows
+#### [Future KEP Consideration in 1.35] Support for Windows
 
 Pod-level resource specifications are a natural extension of Kubernetes' existing
 resource management model. Although this new feature is expected to function with
 Windows containers, careful testing and consideration are required due to
 platform-specific differences. As the introduction of pod-level resources is a
 major change in itself, full support for Windows will be addressed in future
-stages, beyond the initial alpha release.
+KEPs, beyond the scope of this KEP.
+
+#### [Future KEP Consideration in 1.35] Memory Manager
+Please note: This will be a separate feature, as the required changes are
+not trivial and warrant further discussion.
+
+The Memory Manager currently allocates memory resources at
+the container level through its
+[Allocate](https://github.com/kubernetes/kubernetes/blob/849a82b727b1cc1e77b58149b3cacbfa5ada30fd/pkg/kubelet/cm/memorymanager/memory_manager.go#L261)
+method. The [Topology Manager](https://github.com/kubernetes/kubernetes/blob/fd53f7292c7d5899135fddd928c0dc3844126820/pkg/kubelet/cm/topologymanager/scope.go#L150) calls this Allocate method as part of its hint provider integration.
+
+
+With the introduction of Pod Level Resources, the following modifications are needed:
+
+1. Memory Manager Interface Extension:
+Add a new AllocatePodLevel method to the Memory Manager interface to handle 
+resource allocation at the pod level. This method will complement the existing container-level Allocate method.
+
+2. Topology Manager Integration: Modify the (Topology Manager)[https://github.com/kubernetes/kubernetes/blob/fd53f7292c7d5899135fddd928c0dc3844126820/pkg/kubelet/cm/topologymanager/scope.go#L150] to conditionally
+call AllocatePodLevel when pod-level resources are configured. Maintain
+backward compatibility by continuing to use the existing Allocate method for
+container-level allocation scenarios
+
+#### [Future KEP Consideration] CPU Manager
```

Comment:
Why is memory manager targeted for support in v1.35 but CPU manager is not? I'm not sure it makes sense to support memory manager without CPU manager.

Reply:
cc @pravk03, we had a related discussion around in-place resize

Reply:
Typo :|